src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -554,8 +532,8 @@ private void EmitModule(JsonTextWriter memoryWriter, ModuleSymbol moduleSymbol,
                 memoryWriter.WritePropertyName("template");
                 {
                     var moduleSemanticModel = GetModuleSemanticModel(moduleSymbol);
-                    var moduleWriter = new TemplateWriter(memoryWriter, moduleSemanticModel, this.assemblyFileVersion);
-                    moduleWriter.WriteModule();
+                    var moduleWriter = new TemplateWriter(moduleSemanticModel, this.assemblyFileVersion);
+                    moduleWriter.Write(memoryWriter);
```

Comment:
Could we rename `memoryWriter` -> `jsonWriter`? Just to give a clearer idea of what it is doing